packages/salesforcedx-utils/src/cli/checkCliVersion.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { SfdxCommandBuilder } from './sfdxCommandBuilder';
+import { CliCommandExecutor } from './cliCommandExecutor';
+import { CommandOutput } from './commandOutput';
+
+export class CheckCliVersion {
+
+  public async validateCliVersion(): Promise<string> {
+
+    // Execute the command "sfdx --version" in the Terminal
+    const execution = new CliCommandExecutor(
+      new SfdxCommandBuilder()
+        .withArg('--version')
+        .withJson()
+        .build(),
+        {}
+      ).execute();
+
+    // Save the result of the command
+    const cmdOutput = new CommandOutput();
+    const result = await cmdOutput.getCmdResult(execution);
+
+    // Parse the result to check if the version is supported
+    const sfdxPattern = /sfdx-cli\/(\d+\.\d+\.\d+)/;
+    const sfPattern = /@salesforce\/cli\/(\d+\.\d+\.\d+)/;
+
+    const sfdxMatch = sfdxPattern.exec(result);
+    const sfMatch = sfPattern.exec(result);
+
+    // Case 1: SFDX CLI is installed
+    if (sfdxMatch) {
+      const sfdxVersion = sfdxMatch[1];
+      const sfdxVersionNumber = sfdxVersion.split('.').map(Number);
+      // The last working version of SFDX is v7.193.2
+      if (sfdxVersionNumber[0] >= 7 && sfdxVersionNumber[1] >= 193 && sfdxVersionNumber[2] >= 2) {
+        return 'validCli';
+      }
+      else {
+        return 'cliNotSupported';
+      }
+    }
+
+    // Case 2: SF CLI is installed
+    else if (sfMatch) {
+      const sfVersion = sfMatch[1];
+      const sfVersionNumber = sfVersion.split('.').map(Number);
+      // SF v1 does not map sf commands to sfdx commands and is not supported
+      if (sfVersionNumber[0] >= 2 && sfVersionNumber[1] >= 0 && sfVersionNumber[2] >= 0) {
+        return 'validCli';
+      }
+      else {
+        return 'cliNotSupported';
+      }
+    }
+
+    // Case 3: Neither SFDX CLI nor SF CLI is installed
+    else {
```

Comment:
Yeah I'd def go with 
`} else { // Case 3: ......`

Reply:
@gbockus-sf I just ran Prettier against that file and Prettier actually didn't complain about the if/else with spaces in between.  I tried switching to the formatting that you suggested in this comment but when I saved the file, Prettier moved the comment to the next line (see attached screenshot for Option 3).  What would you prefer that I do?

1. Keep the original formatting with the spaces between the if/else statements
![Screenshot 2023-11-09 at 4 48 00 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/dfe05980-27d5-4f2f-95ec-000de70b43ee)

2. Turn off Prettier for the file and reformat to what you suggested
![Screenshot 2023-11-09 at 4 49 09 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/cb47dee5-826e-42db-acb4-0828cd7e5f3b)

3. Let Prettier auto-format the comments and keep them inside the brackets on the next line
![Screenshot 2023-11-09 at 4 42 34 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/9a3e8649-0eca-4734-b0a6-5e78cb86cd93)

Reply:
I like either 1 or 3 🙂